microsoft-edge/web-platform/translator-api.md

```diff
@@ -0,0 +1,391 @@
+
+1. Optionally, to log information locally that may be useful for debugging issues, also enable the **Enable on device AI model debug logs** flag.
+
+1. Restart Microsoft Edge Canary or Dev.
+
+
+<!-- ====================================================================== -->
+## See a working example
+
+To see the Translator API in action, and review existing code that uses these APIs:
+
+1. [Enable the Translator API](#enable-the-translator-api), as described above.
+
+1. In Microsoft Edge Canary or Dev, go to the [Translator API playground](https://microsoftedge.github.io/Demos/built-in-ai/playgrounds/translator-api/) in a new window or tab.
+
+1. In the information banner at the top, check the status: it initially reads: **On-device API and model downloadable. The model for a specified language pair will be downloaded the first time the API is used.**
+
+1. In the **Text to translate** text box, enter text to translate, and optionally change the **Source language** and **Target language** values.
+
+1. Click the **Translate** button.
+
+   If the model for the specified language pair hasn't been downloaded before, the download starts.
+   
+   ![Status indicator showing model downloading progress](./translator-api-images/model-downloading.png)
+   
+   If the model download doesn't start, restart Microsoft Edge and try again.
+
+   After the model has downloaded, the model starts generating a translation of the text from the source language into the target language.
+
+   The output is generated in the response section of the page:
+
+   ![The Translator demo page with settings and a Translate button](./translator-api-images/translating.png)
+
+1. To stop generating the response, at any time, click the **Stop** button.
+
+See also:
+* [/built-in-ai/](https://github.com/MicrosoftEdge/Demos/tree/main/built-in-ai/) - Source code and Readme for the Built-in AI playgrounds demo.
+
+
+<!-- ====================================================================== -->
+## Use the Translator API
+<!-- leverage high-value h2s below -->
+
+The next sections are about using the Translator API.
+
+
+<!-- ====================================================================== -->
+## Check if the Translator API is enabled
+
+Before using the Translator API in your website's code, check that the API is enabled by testing the presence of the `Translator` object:
+
+```javascript
+if (!Translator) {
+  // The Translator API is not available.
+} else {
+  // The Translator API is available.
+}
+```
+
+
+<!-- ====================================================================== -->
+## Check if the model can be used (`availability()`)
+
+The Translator API has the following requirements:
+
+* The device must support running the model.
+* The language model and the model runtime must already have been downloaded by Microsoft Edge.
+* Translating from the specified source language to the specified target language must be supported by the model.
+
```

Comment:
NOTE: The API uses [BCP 47](linkto: https://www.rfc-editor.org/info/bcp47) language codes as strings. For example, 'en' for English or 'es' for Spanish.